metamngmt/msgtypes.go

```diff
@@ -59,10 +60,13 @@ type AddDerived struct {
 	Nonce []byte `json:"nonce"`
 
 	Tangles refs.Tangles `json:"tangles"`
+	QueryLang string `json:"querylang"`
+	Query string `json:"query"`
```

Comment:
Thinking about this some more.. I think it's not good to export them like this.

With this approach can't differentiate the absence of the fields in the bencode data from empty string values.

```suggestion
      metadata map[string]string
```

Plus a function ala `HasMetadata(key string) (string, bool)`